lib/adapter.js

```diff
@@ -1137,8 +1139,13 @@ function Adapter(options) {
                     if (!obj.common.hasOwnProperty('type')) {
                         logger.warn(that.namespace + ' setObject ' + id + ' (type=' + obj.type + ') property common.type missing!');
                     }
+                    if (obj.common.hasOwnProperty('def') && obj.common.def !== undefined) {

```

Comment:
This could just be `if (obj.common.def != undefined)`. `!=` checks for `null` and `undefined` as opposed to `!==`

Reply:
Thanks, will adjust it this evening.

Reply:
Although... not sure if `null` is a valid default value? Then it should just be `!== undefined`